src/metabase/pulse.clj

```diff
@@ -80,29 +83,73 @@
     (compare (:row dashcard-1) (:row dashcard-2))
     (compare (:col dashcard-1) (:col dashcard-2))))
 
-(defn- virtual-card-of-type?
-  "Check if dashcard is a virtual with type `ttype`.
-
-  There are currently 3 types of virtual card: text, action, link."
-  [dashcard ttype]
-  (and (map? dashcard)
-       (= ttype (get-in dashcard [:visualization_settings :virtual_card :display]))))
+(defn- link-card->url
+  [{:keys [entity url] :as link-card}]
+  (if url
+    (:url link-card)
+    (let [{:keys [db_id id model]} entity]
+      (case model
+        "card"       (urls/card-url id)
+        "dataset"    (urls/card-url id)
+        "collection" (urls/collection-url id)
+        "dashboard"  (urls/dashboard-url id)
+        "database"   (urls/database-url id)
+        "table"      (urls/table-url db_id id)))))
+
+(defn- link-card->content
+  [{:keys [entity url] :as _link-card}]
+  (if url
+    url
+    (:name entity)))
+
+(defn- link-card->description
+  [{:keys [entity url] :as _link-card}]
+  (if url
+    nil
+    (:description entity)))
```

Comment:
I was trying to be explicit because the schema of the entity and URL are different. yes it could be shortened but I think readability is important in this case.